crates/polars-ops/src/series/ops/is_last.rs

```diff
@@ -0,0 +1,173 @@
+            is_last_bin(ca)
+        },
+        Utf8 => {
+            let s = s.cast(&Binary).unwrap();
+            return is_last(&s);
+        },
+        Float32 => {
+            let ca = s.bit_repr_small();
+            is_last_numeric(&ca)
+        },
+        Float64 => {
+            let ca = s.bit_repr_large();
+            is_last_numeric(&ca)
+        },
+        dt if dt.is_numeric() => {
+            with_match_physical_integer_polars_type!(s.dtype(), |$T| {
+                let ca: &ChunkedArray<$T> = s.as_ref().as_ref().as_ref();
+                is_last_numeric(ca)
+            })
+        },
+        #[cfg(feature = "dtype-struct")]
+        Struct(_) => return is_last_struct(&s),
+        #[cfg(feature = "group_by_list")]
+        List(inner) if inner.is_numeric() => {
+            let ca = s.list().unwrap();
+            return is_last_list(ca);
+        },
+        dt => polars_bail!(opq = is_last, dt),
+    };
+    Ok(out)
+}
+
+fn is_last_boolean(ca: &BooleanChunked) -> BooleanChunked {
+    let mut out = MutableBitmap::with_capacity(ca.len());
+    out.extend_constant(ca.len(), false);
+
+    if ca.null_count() == ca.len() {
+        out.set(ca.len() - 1, true);
+    }
+    // TODO supports fast path.
+    else {
+        let mut first_true_found = false;
+        let mut first_false_found = false;
+        let mut first_null_found = false;
+        let mut all_found = false;
+        ca.into_iter()
+            .enumerate()
+            .rev()
+            .find_map(|(idx, val)| match val {
+                Some(true) if !first_true_found => {
+                    first_true_found = true;
+                    all_found &= first_true_found;
+                    out.set(idx, true);
+                    if all_found {
+                        Some(())
+                    } else {
+                        None
+                    }
+                },
+                Some(false) if !first_false_found => {
+                    first_false_found = true;
+                    all_found &= first_false_found;
+                    out.set(idx, true);
+                    if all_found {
+                        Some(())
+                    } else {
+                        None
+                    }
+                },
+                None if !first_null_found => {
+                    first_null_found = true;
+                    all_found &= first_null_found;
+                    out.set(idx, true);
+                    if all_found {
+                        Some(())
+                    } else {
+                        None
+                    }
+                },
+                _ => None,
+            });
+    }
+
+    let arr = BooleanArray::new(ArrowDataType::Boolean, out.into(), None);
+    BooleanChunked::with_chunk(ca.name(), arr)
+}
+
+fn is_last_bin(ca: &BinaryChunked) -> BooleanChunked {
+    let mut unique = PlHashSet::new();
+    let mut new_ca: BooleanChunked = ca
+        .into_iter()
```

Comment:
Initially, I impl this based on `downcast_iter + arr.into_iter`. But that way requires us to collect both the inner and outer iterator all collected by reverse order. For the inner loop, I just introduce `impl FromIteratorReversed<bool> for BooleanArray`. But for the outer loop, we also need to make `ChunkedArray::from_chunk_iter` supports reverse collection, but I'm not sure if this is suitable, so I have temporarily adopted a non nested approach.

Reply:
Maybe we can rechunk and then iterate over the arrow array.

This should make iteration fast for the most common case.